### PR TITLE
fix: send request XML object instead of Object.toString() to execute()

### LINKED ISSUE #881: Unable to create bucket when updating java client from 6.x.x to 7.0.0
Error:
`
error occurred
ErrorResponse(code = MalformedXML, message = The XML you provided was not well-formed or did not validate against our published schema.
, bucketName = my-test-bucket, objectName = null, resource = /my-test-bucket, requestId = 15FFDB8A67106604, hostId = 96fa3866-7ee6-4546-87d9-4283e3def6c3)
request={method=PUT, url=https://play.min.io/my-test-bucket, headers=Host: play.min.io
`

Code:

https://github.com/testcontainers/testcontainers-spring-boot/blob/develop/embedded-minio/src/test/java/com/playtika/test/minio/EmbeddedMinioBootstrapConfigurationTest.java#L43

Change:
https://github.com/testcontainers/testcontainers-spring-boot/pull/228

Build results:
https://circleci.com/gh/testcontainers/testcontainers-spring-boot/1020?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

### PR DESCRIPTION
**Pull request by balamurugana**
Fixes #881

### CODE REVIEW

**Comment by kannappanr** [comment]
```
import io.minio.MinioClient;
import io.minio.errors.MinioException;
import java.io.IOException;
import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;
import org.xmlpull.v1.XmlPullParserException;

public class MakeBucket {
  /** MinioClient.makeBucket() example. */
  public static void main(String[] args)
      throws IOException, NoSuchAlgorithmException, InvalidKeyException, XmlPullParserException {
    try {
      /* Amazon S3: */
      MinioClient minioClient = new MinioClient("https://s3.amazonaws.com", "YOUR-ACCESSKEYID", "YOUR-SECRETACCESSKEY");

      // Create bucket if it doesn't exist.
      boolean found = minioClient.bucketExists("non-existent-bucket");
      minioClient.traceOn(System.out);
      if (found) {
        System.out.println("non-existent-bucket already exists");
      } else {
        // Create bucket 'non-existent-bucket'.
        minioClient.makeBucket("non-existent-bucket", "us-west-1", false);
        System.out.println("non-existent-bucket is created successfully");
      }
    } catch (MinioException e) {
      System.out.println("Error occurred: " + e);
    }
  }
}

```

File: api/src/main/java/io/minio/MinioClient.java [hunk 1]
```diff
@@ -3301,13 +3301,11 @@ public void makeBucket(String bucketName, String region, boolean objectLock)
...
-      region = US_EAST_1;
-      configString = "";
-    } else {
+
+    String configString = "";
+    // Location constraint is not required for 'us-east-1' region. Refer
+    // http://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region
+    if (region != null && !region.equals(US_EAST_1)) {
       CreateBucketConfiguration config = new CreateBucketConfiguration(region);
       configString = config.toString();
```
Comment (harshavardhana) [inline api/src/main/java/io/minio/MinioClient.java]: @balamurugana I think the issue is `config.toString()` which is not returning XML here

Comment (balamurugana) [reply api/src/main/java/io/minio/MinioClient.java]: Yep. Fixed.

**Comment by kannappanr** [comment]
This still doesn't work when you set `MINIO_REGION=us-west-1` and start minio server and use the code provided above

**Comment by balamurugana** [comment]
> This still doesn't work when you set `MINIO_REGION=us-west-1` and start minio server and use the code provided above

Can you enable `MinioClient.traceOn()` and share the trace?

**Comment by balamurugana** [comment]
> This still doesn't work when you set `MINIO_REGION=us-west-1` and start minio server and use the code provided above

Minio server master doesn't work with region. `minio-go` throws `2020/03/28 13:23:11 The authorization header is malformed; the region is wrong; expecting 'us-east-1'.`

**Comment by harshavardhana** [comment]
If the server region is set then it's like AWS region specific endpoints, so you can't use us-east-1 for that minio-go constructor needs to be initialized with the right region.

**Comment by balamurugana** [comment]
> If the server region is set then it's like AWS region specific endpoints, so you can't use us-east-1 for that minio-go constructor needs to be initialized with the right region.

Found actual behaviour and fixed

**Pull request merged**